Adding paper.md for JOSS (#110)
cosmo-epfl/scikit-cosmo: paper.md

which represent atoms is almost unbounded, and it becomes necessary to assess
their utility, remove redundancies, or condense them. Currently, `scikit-COSMO`
contains algorithms to address these specific problems, all of which were
previously-unavailable within open-source repositories.

- Feature Reconstruction Measures[@Goscinski2021] - both global and local measures
of the information density for a given representation. This set of
unsupervised algorithms is of particular use in the choice of representation
and hyperparameter optimization.
- Feature and Sample Selection[@Imbalzano2018
                               @Cersonsky2021] - methods focused on determining
a diverse or information-rich subset of features or samples for machine
learning problems.
- Partially-Supervised Dimensionality Reduction[@deJong1992
                                                @Helfrecht2020] - linear and
non-linear techniques to combine features into latent-space projections
(similar to PCA) that also incorporate target information. This is of particular
use when condensing features prior to property regression or constructing

> structure-property maps, such as those analyzable via[@Fraux2020].

Reviewer: I just want to quickly add here that I don't think we should describe PCovR-based methods as "partially-supervised" -- I think it could easily be confused with "semi-supervised" machine learning methods, which is something completely different from what PCovR does (typically doing supervised learning with missing labels on some of the training samples). I would instead opt for "Combined Supervised-Unsupervised Dimensionality Reduction" or "Hybrid Supervised-Unsupervised Dimensionality Reduction". Since PCovR also does regression, it's also more than just dimensionality reduction, and can also be considered a different type of "regularization" to deal with multicollinearity.

Author: Amended! Let me know how it reads.

---
title: 'scikit-COSMO: A Toolbox of Machine Learning Methods for Materials Science'
tags:
    - Python
authors:
    - name: Rose K. Cersonsky ^ [\url{mailto: rose.cersonsky@epfl.ch}]
    orcid: 0000-0003-4515-3441
    affiliation: 1
    - name: Guillaume Fraux
    orcid: 0000-0003-4824-6512
    affiliation: 1
    - name: Sergei Kliavinek
    orcid: 0000-0001-8326-325X
    affiliation: 1
    - name: Alexander Goscinski
    orcid: 0000-0001-8076-215X
    affiliation: 1
    - name: Benjamin A. Helfrecht
    orcid: 0000-0002-2260-7183
    affiliation: 1
    - name: Michele Ceriotti ^ [\url{mailto: michele.ceriotti@epfl.ch}]
    orcid: 0000-0003-2571-2832
    affiliation: 1
affiliations:
    - name: Laboratory of Computational Science and Modeling(COSMO), École Polytechnique Fédérale de Lausanne(EPFL), Lausanne, Switzerland
    index: 1
date: 03 May 2021
bibliography: paper.bib

> 

Reviewer 2: maybe this empty line is causing the metadata issue with the preview?

Author: The date line?

Reviewer 2: no, the empty one after bibliography

---

> # Summary

Reviewer 2: this is missing a "we use Python for this" and maybe a "We use Python for this, because XXX"

Author: How about now?

> `scikit-COSMO` (skcosmo) focuses on machine learning methods that are of particular use(but not limited

Reviewer 2: ```suggestion
`scikit-COSMO` (skcosmo) focuses on machine learning methods that are of particular use (but not limited
```

                                                                                        to) the field of materials science and computational chemistry. Currently, many

> machine learning studies employ the foundational `scikit-learn` repository, a

Reviewer 2: ```suggestion
machine learning studies in computational chemistry employ the foundational `scikit-learn` repository, a
```

Author: More than just in computational chemistry. scikit-learn is used pretty universally

Reviewer 2: Since the sentences around are specifically referring to this field, maybe something like:

> machine learning studies across fields of science employ the foundational `scikit-learn` repository, a

Author: The point I'm trying to make is the scikit-learn is intended for near-universal machine learning problems, and so it may not pay sufficient attention to some ML components as is necessitated within our field. Is that not coming across well?

collection of widely applicable machine learning algorithms and methods.
Written in the same style and guidelines as `scikit-learn`, we aim to provide

> users the ability to seamlessly include these methods in machine learning

Reviewer 2: ```suggestion
users the ability to seamlessly include methods tailored for material science in machine learning
```

workflows alongside those from `scikit-learn`.

# Statement of need

> In `scikit-COSMO`, we provide machine learning algorithms developed for a specific

Reviewer 2: I prefer moving the sentence around a bit (I don't know the proper gramatical name for this 😃 ), but feel free to disagree

```suggestion
The `scikit-COSMO` package provides machine learning algorithms developed for a specific
```

community: those who use machine learning to understand or represent collections
of atoms or molecules. The emphasis put upon each subclass of machine learning
algorithms varies within different subfields. For example, in applying machine
learning to collections of atoms, there is a necessary focus on the
representation of data in a numerical format for machine learning pipelines,
known as _featurization_. Many machine learning studies represent samples using
real-world measurements or tests, effectively limiting the number of features
based upon the resources available. When representing a collection of atoms,

> there is both a variety of featurization techniques to employ and a wealth of

Reviewer 2: We could add a small list of such representation & parameters, and link to the tree of representation paper here.

which represent atoms is almost unbounded, and it becomes necessary to assess
their utility, remove redundancies, or condense them. Currently, `scikit-COSMO`
contains algorithms to address these specific problems, all of which were
previously-unavailable within open-source repositories.

- Feature Reconstruction Measures[@Goscinski2021] - both global and local measures
of the information density for a given representation. This set of
unsupervised algorithms is of particular use in the choice of representation
and hyperparameter optimization.
- Feature and Sample Selection[@Imbalzano2018
                               @Cersonsky2021] - methods focused on determining
a diverse or information-rich subset of features or samples for machine
learning problems.
- Partially-Supervised Dimensionality Reduction[@deJong1992
                                                @Helfrecht2020] - linear and
non-linear techniques to combine features into latent-space projections
(similar to PCA) that also incorporate target information. This is of particular
use when condensing features prior to property regression or constructing
structure-property maps, such as those analyzable via[@Fraux2020].

> Note: `scikit-COSMO` is not intended to compute atomic descriptors, as this is
> the role of the companion package `librascal` [@Musil2021] or similar, such as
> QUIP[@quip].

Reviewer 2: ```suggestion
`scikit-COSMO` itself does not compute atomic descriptors directly, instead taking as input matrices of descriptor computed by other software such as `librascal` [@Musil2021],
`QUIP`[@quip], `DScribe`[https://github.com/SINGROUP/dscribe] and similar software.
```

Author: ```suggestion
Note: `scikit-COSMO` itself does not compute atomic descriptors directly, and instead takes as input descriptors compute by other software such as the companion package `librascal` [@Musil2021] or similar, including
QUIP[@quip] and `DScribe`[https://github.com/SINGROUP/dscribe].
```

We should highlight the increased compatibility that will *hopefully* be achieved with librascal.

Reviewer 2: > We should highlight the increased compatibility that will hopefully be achieved with librascal.

I disagree, we should strive for good compatibility with more than just software we are developing ourself, because this makes skcosmo useful to more people.

Author: Fair point. How about this?
```suggestion
Note: `scikit-COSMO` itself does not compute atomic descriptors directly, and instead takes as input descriptors computed by other software such as `librascal` [@Musil2021],
QUIP[@quip], and `DScribe`[https://github.com/SINGROUP/dscribe].
```

Reviewer 2: That sounds good to me 👍 . I don't like the `Note:` prefix, but if you want to keep I'll be fine!

Author: Bof, it's an idiosyncrasy that I picked up from @ceriottm, and a habit I'm trying to break :D

> scikit-COSMO also contains minimal datasets for module testing, including a small

Reviewer 2: ```suggestion
scikit-COSMO also contains minimal datasets used to test the implementation, including a small
```

                               @Cersonsky2021] - methods focused on determining
a diverse or information-rich subset of features or samples for machine
learning problems.

> - Hybrid Supervised-Unsupervised Dimensionality Reduction and Regression d[@deJong1992

Reviewer: ```suggestion
- Hybrid Supervised-Unsupervised Dimensionality Reduction and Regression[@deJong1992
```
